PHPStan level 9 failure

**PHP version**: 7.4 

**Description**
Running PHPStan level 9 checks on this library fails with

```
 ------ -----------------------------------------------------------------------------------------------
  Line   FulfilledPromise.php
 ------ -----------------------------------------------------------------------------------------------
  58     Method Http\Promise\FulfilledPromise::wait() should return T but return statement is missing.
 ------ -----------------------------------------------------------------------------------------------

 ------ ----------------------------------------------------------------------------------------------
  Line   RejectedPromise.php
 ------ ----------------------------------------------------------------------------------------------
  55     Method Http\Promise\RejectedPromise::wait() should return T but return statement is missing.
 ------ ----------------------------------------------------------------------------------------------


 [ERROR] Found 2 errors

```

**How to reproduce**
`phpstan analyze --level 9 src/`

**Possible Solution**
Expand return type of `wait()` to accommodate `void` and `null` return types

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -60,3 +60,18 @@ jobs:
         run: |
           wget https://scrutinizer-ci.com/ocular.phar
           php ocular.phar code-coverage:upload --format=php-clover build/coverage.xml
+
+  phpstan:
+    name: PHPStan
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: PHPStan
+        uses: OskarStark/phpstan-ga@0.12.32
+        env:
+          REQUIRE_DEV: false
+        with:
+          args: analyze --no-progress

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Change Log
 
+## 1.2.1
+
+### Added
+
+- Fixed PHPDoc for `wait()` and `then()`'s `onRejected` callable
+
 ## 1.2.0 - 2023-10-24
 
 ### Added

--- a/phpstan.neon.dist
+++ b/phpstan.neon.dist
@@ -1,0 +1,6 @@
+parameters:
+    level: max
+    checkMissingIterableValueType: false
+    treatPhpDocTypesAsCertain: false
+    paths:
+        - src

--- a/src/FulfilledPromise.php
+++ b/src/FulfilledPromise.php
@@ -58,6 +58,7 @@ final class FulfilledPromise implements Promise
         if ($unwrap) {
             return $this->result;
         }
+
         return;
     }
 }

--- a/src/FulfilledPromise.php
+++ b/src/FulfilledPromise.php
@@ -58,5 +58,6 @@ final class FulfilledPromise implements Promise
         if ($unwrap) {
             return $this->result;
         }
+        return;
     }
 }

--- a/src/Promise.php
+++ b/src/Promise.php
@@ -65,7 +65,7 @@ interface Promise
      *
      * @param bool $unwrap Whether to return resolved value / throw reason or not
      *
-     * @return T|void|null Resolved value, null if $unwrap is set to false
+     * @return ($unwrap is true ? T : null|void) Resolved value, null if $unwrap is set to false
      *
      * @throws \Exception the rejection reason if $unwrap is set to true and the request failed
      */

--- a/src/Promise.php
+++ b/src/Promise.php
@@ -65,7 +65,7 @@ interface Promise
      *
      * @param bool $unwrap Whether to return resolved value / throw reason or not
      *
-     * @return ($unwrap is true ? T : null|void) Resolved value, null if $unwrap is set to false
+     * @return ($unwrap is true ? T : null) Resolved value, null if $unwrap is set to false
      *
      * @throws \Exception the rejection reason if $unwrap is set to true and the request failed
      */

--- a/src/Promise.php
+++ b/src/Promise.php
@@ -39,7 +39,7 @@ interface Promise
      * The callback will be called when the value arrived and never more than once.
      *
      * @param callable(T): V|null          $onFulfilled called when a response will be available
-     * @param callable(\Exception): V|null $onRejected  called when an exception occurs
+     * @param callable(\Throwable): V|null $onRejected  called when an exception occurs
      *
      * @return Promise<V> a new resolved promise with value of the executed callback (onFulfilled / onRejected)
      *
@@ -65,7 +65,7 @@ interface Promise
      *
      * @param bool $unwrap Whether to return resolved value / throw reason or not
      *
-     * @return T Resolved value, null if $unwrap is set to false
+     * @return T|void|null Resolved value, null if $unwrap is set to false
      *
      * @throws \Exception the rejection reason if $unwrap is set to true and the request failed
      */

--- a/src/RejectedPromise.php
+++ b/src/RejectedPromise.php
@@ -55,5 +55,6 @@ final class RejectedPromise implements Promise
         if ($unwrap) {
             throw $this->exception;
         }
+        return;
     }
 }

--- a/src/RejectedPromise.php
+++ b/src/RejectedPromise.php
@@ -55,6 +55,7 @@ final class RejectedPromise implements Promise
         if ($unwrap) {
             throw $this->exception;
         }
+
         return;
     }
 }

